src/encoded/static/components/gene_search/match.js

```diff
@@ -0,0 +1,108 @@
+import {
+    MATCH,
+    MISMATCH,
+} from './constants';
+
+
+/**
+* Find all of the start indices of a pattern in a text string.
+* @param {string} pattern - The pattern to search for in the string.
+* @param {string} text - The string which may contain the pattern.
+* @return {array} The start indices of the pattern in the text.
+*/
+export const getIndicesOf = (pattern, text) => {
+    const indices = [];
+    let currentIndex = 0;
+    const patternLength = pattern.length;
+    // We check patternLength to avoid infinite loop if user passes in
+    // empty string. Otherwise we set the currentIndex while checking
+    // the condition on every loop.
+    /* eslint no-cond-assign: ["error", "except-parens"] */
```

Comment:
https://eslint.org/docs/rules/no-cond-assign#options

Maybe I'm using this wrong, the rule seems to have a default string option that `allows assignments in test conditions only if they are enclosed in parentheses (for example, to allow reassigning a variable in the test of a while or do...while loop)`. And the assignment is in parenthesis: `(currentIndex = text.indexOf(pattern, currentIndex))`.

It seems to work if I don't include the `except-parens` in the `eslint no-cond-assign` list, since that's the default, but I was trying to make it explicit.

In this case I think it's one of the `valid reasons to use assignment operators in conditional statements` since it avoids either checking the same indexOf twice per loop or having a duplicate indexOf call and extra variable before the loop. 

Is there a reason the `except-parens` option isn't the default for our eslint rules?

Reply:
I think you make a reasonable argument. I’d have no issue if you allowed except-parens in .eslintrc.js so you don’t need the aslant comment.

Reply:
Okay! Think I've updated the rules and removed this line.